### PR TITLE
Add CPU BatchNormalization for NCHW format

### PR DESCRIPTION
**Pull request by kaixih**
Previously, CPU doesn't support NCHW. This makes other operations based on NCHW batch norm hard to use. For example, the layer norm can be transformed to use NCHW batch norm on GPU. However, since we don't have CPU NCHW batch norm, we have to either avoid using NCHW batch norm or add more parameters to control the execution path.

This PR enables NCHW batch norm on CPU.

fyi. @nluehr

### CODE REVIEW

**Comment by houtoms** [comment]
This PR is originally served for https://github.com/tensorflow/tensorflow/pull/31557.

@reedwm

File: tensorflow/core/kernels/fused_batch_norm_op.cc [hunk 1]
```diff
@@ -262,13 +260,38 @@ struct FusedBatchNorm<CPUDevice, T, U> {
...
                                  "does not support activations."));
-    typename TTypes<T, 4>::ConstTensor x(x_input.tensor<T, 4>());
+    Tensor transformed_x;
+    Tensor transformed_y;
+    if (tensor_format == FORMAT_NCHW) {
+      const int64 in_batch = GetTensorDim(x_input, tensor_format, 'N');
+      const int64 in_rows = GetTensorDim(x_input, tensor_format, 'H');
+      const int64 in_cols = GetTensorDim(x_input, tensor_format, 'W');
+      const int64 in_depths = GetTensorDim(x_input, tensor_format, 'C');
+			OP_REQUIRES_OK(context, context->allocate_temp(
```
Comment (reedwm) [inline tensorflow/core/kernels/fused_batch_norm_op.cc]: Please use spaces over tabs. And same for other parts of the file.

File: tensorflow/python/ops/nn_fused_batchnorm_test.py [hunk 2]
```diff
@@ -377,6 +379,8 @@ def testInferenceShape4(self):
             x_shape, dtype, [6], np.float32, use_gpu=True, data_format='NHWC')
       self._test_inference(
           x_shape, dtype, [6], np.float32, use_gpu=False, data_format='NHWC')
+      self._test_inference(
```
Comment (reedwm) [inline tensorflow/python/ops/nn_fused_batchnorm_test.py]: For consistency with the GPU case, you should put the NCHW case first. And same for some of the test cases below

File: tensorflow/python/ops/nn_fused_batchnorm_test.py [hunk 3]
```diff
@@ -404,6 +411,8 @@ def testTrainingShape1(self):
             x_shape, dtype, [1], np.float32, use_gpu=True, data_format='NCHW')
       self._test_training(
           x_shape, dtype, [1], np.float32, use_gpu=False, data_format='NHWC')
+      self._test_training(
+          x_shape, dtype, [1], np.float32, use_gpu=False, data_format='NCHW')
 
   def testTrainingShape2(self):
```
Comment (reedwm) [inline tensorflow/python/ops/nn_fused_batchnorm_test.py]: Can you add a CPU test to the testTrainingShape3 and testBatchNormGradShape3 functions?

**Review by reedwm — CHANGES_REQUESTED** [review]

**Comment by houtoms** [comment]
Done. Please check.

**Review by reedwm — APPROVED** [review]
